receiver/prometheusreceiver/internal/prom_to_otlp_test.go

```diff
@@ -52,13 +52,15 @@ type jobInstanceDefinition struct {
 	job, instance, host, scheme, port string
 }
 
-func makeResourceWithJobInstanceScheme(def *jobInstanceDefinition) pdata.Resource {
+func makeResourceWithJobInstanceScheme(def *jobInstanceDefinition, noHost bool) pdata.Resource {
 	resource := pdata.NewResource()
 	attrs := resource.Attributes()
 	// Using hardcoded values to assert on outward expectations so that
 	// when variables change, these tests will fail and we'll have reports.
 	attrs.UpsertString("service.name", def.job)
-	attrs.UpsertString("host.name", def.host)
+	if !noHost {
```

Comment:
If this flag is used only here, consider making it in the affirmative case, like: `hasHost`. Using a double-negative might be confusing.

Reply:
fixed, it's only used there indeed